### ISSUE TITLE
SIGSEGV happening randomly while running tests.

### ISSUE THREAD

**Issue by kdamevin** [AdoptOpenJDK/openjdk-support#55]
**Platform:**
Java VM: OpenJDK 64-Bit Server VM (11.0.4+11, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)

**Architecture:**
linux-amd64

Hard to reproduce consistently unfortunately but we are seeing the following segmentation fault error popping up intermittently in our test suite.

```
# A fatal error has been detected by the Java Runtime Environment:
#
SUREFIRE-859: #  SIGSEGV (0xb) at pc=0x00007f42e9389120, pid=502, tid=505
#
# JRE version: OpenJDK Runtime Environment (11.0.4+11) (build 11.0.4+11)
SUREFIRE-859: # Java VM: OpenJDK 64-Bit Server VM (11.0.4+11, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# J 37159 c2 org.springframework.boot.context.properties.source.ConfigurationPropertyName.defaultElementEquals(Lorg/springframework/boot/context/properties/source/ConfigurationPropertyName$Elements;Lorg/springframework/boot/context/properties/source/ConfigurationPropertyName$Elements;I)Z (436 bytes) @ 0x00007f42e9389120 [0x00007f42e93890c0+0x0000000000000060]
#
SUREFIRE-859: # No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /opt/company/pipelines/agent/build/hs_err_pid502.log

Could not load hsdis-amd64.so; library not loadable; PrintAssembly is disabled
```

See  attached hs_err log file [hs_err_pid502.log](https://github.com/AdoptOpenJDK/openjdk-build/files/4183782/hs_err_pid502.log)

**Comment by kdamevin** [comment]
hey @karianna, yes I just managed to have it happening also with 11.0.6 (Also noticed it happens on 12.0.2+10). Unfortunately I am still not able to reproduce consistency/locally, just on the build server.

```SUREFIRE-859: #  SIGSEGV (0xb) at pc=0x00007f5ab5705a1e, pid=539, tid=542
#
# JRE version: OpenJDK Runtime Environment (11.0.6+10) (build 11.0.6+10)
SUREFIRE-859: # Java VM: OpenJDK 64-Bit Server VM (11.0.6+10, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# J 42409 c2 org.springframework.boot.context.properties.source.ConfigurationPropertyName.defaultElementEquals(Lorg/springframework/boot/context/properties/source/ConfigurationPropertyName$Elements;Lorg/springframework/boot/context/properties/source/ConfigurationPropertyName$Elements;I)Z (436 bytes) @ 0x00007f5ab5705a1e [0x00007f5ab57059c0+0x000000000000005e]
#
SUREFIRE-859: # No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /opt/company/pipelines/agent/build/hs_err_pid539.log
Compiled method (c2)  273821 22860       4       java.util.LinkedHashMap::get (33 bytes)
Could not load hsdis-amd64.so; library not loadable; PrintAssembly is disabled```

**Comment by grzesuav** [comment]
@kdamevin seems related to https://stackoverflow.com/questions/53319466/spring-boot-build-fails-error-when-starting-fork-exit-code-134 ?

**Comment by grzesuav** [comment]
I noticed that you are also using jacoco instrumentation, as suggested in stackoverflow maybe you can remove it temporarily and  see if the problem lies there ?

**Comment by ullahfaheem** [comment]
I have the exact same problem. Everything works with `OpenJDK 8.0.222`. I tried with `OpenJDK Runtime Environment (11.0.5+10) (build 11.0.5+10)` and `12` as well. Consistently breaks with the same error. 

I think the problem is caused by [jacoco](https://docs.gradle.org/current/userguide/jacoco_plugin.html). The problem goes away if I remove all traces of that plugin.

**Comment by grzesuav** [comment]
@ullahfaheem have you reported issue on jacoco ?

**Comment by ullahfaheem** [comment]
No, I haven't yet. I wanted to run some more tests before posting there... but I think it's pretty clear now. The problem is caused by that plugin. I'll post it there soon.

**Comment by chipkillmar** [comment]
We're encountering the same segfault running with OpenJDK `11.0.7+10`.

```
16:45:51 # A fatal error has been detected by the Java Runtime Environment:
16:45:51 #
16:45:51 #  SIGSEGV (0xb) at pc=0x00007f67511e421e, pid=20953, tid=20978
16:45:51 #
16:45:51 # JRE version: OpenJDK Runtime Environment (11.0.7+10) (build 11.0.7+10)
16:45:51 # Java VM: OpenJDK 64-Bit Server VM (11.0.7+10, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)
16:45:51 # Problematic frame:
16:45:51 # J 28853 c2 org.springframework.boot.context.properties.source.ConfigurationPropertyName.defaultElementEquals(Lorg/springframework/boot/context/properties/source/ConfigurationPropertyName$Elements;Lorg/springframework/boot/context/properties/source/ConfigurationPropertyName$Elements;I)Z (436 bytes) @ 0x00007f67511e421e [0x00007f67511e41c0+0x000000000000005e]
```

**Comment by aahlenst** [comment]
@chipkillmar and @kdamevin: If you want us to do anything about it, we need multiple things from you.

* An executable test case that we can run, preferably a link to a public GitHub repository.
* Verification that it happens with https://adoptopenjdk.net/upstream.html (8 or 11) or http://jdk.java.net (14) if one of those is available for your platform.

**Comment by chipkillmar** [comment]
@aahlenst thanks.  Replicating the crash is nontrivial, and it only happens occasionally.  Running without the JaCoCo agent, or with the agent and the `-Xint` option, prevents the crash from happening.

I reported the crash above in version `11.0.7+10` which appears to be the latest upstream version available in the link you posted.  Please let me know if I'm missing something.

**Comment by karianna** [comment]
@chipkillmar I think this needs to be reported to the jacoco folks - Marc is pretty responsive over there.

**Comment by karianna** [comment]
@chipkillmar Ah thank you - reread the thread and this should go upstream to OpenJDK, I'll request a reporter

**Comment by aahlenst** [comment]
@chipkillmar What about trying with [Petclinic](https://github.com/spring-projects/spring-petclinic) and jacoco? Or the entire Spring Boot test suite? Even if it happens 1 in 10 times, it's better than nothing. Otherwise, it's very likely this is going to go nowhere because we have to convince a OpenJDK committer to report it.

**Comment by chipkillmar** [comment]
@aahlenst I can replicate this in my development environment, but I'm running [into this issue](https://github.com/AdoptOpenJDK/openjdk-support/issues/98) where OpenJDK isn't creating a core file after crashing.

**Comment by aahlenst** [comment]
A reproducer alone should be sufficient.

**Comment by aahlenst** [comment]
@chipkillmar If you still have the issue and can share steps to reproduce, please do so. Otherwise I'm going to close this issue because we cannot work on it without further info.

**Comment by chipkillmar** [comment]
@aahlenst I don't have code replicating the crash that I'm at liberty to share.  You may close this.

**Comment by mvmn** [comment]
Got same issue on AdoptOpenJDK 11.0.7+10
Unfortunately, I cannot share the code or memory dumps for reproducing the issue since it's proprietary.

**Comment by karianna** [comment]
> Got same issue on AdoptOpenJDK 11.0.7+10
> Unfortunately, I cannot share the code or memory dumps for reproducing the issue since it's proprietary.

Can you Try 11.0.10+9

**Comment by rmpestano** [comment]
Just to let you know, I had the same issue on `OpenJDK 64-Bit Server VM (11.0.4+11, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)` and fixed by upgrading to `11.0.10`
